projects/batfish/src/main/java/org/batfish/representation/juniper/JuniperConfiguration.java

```diff
@@ -247,6 +258,28 @@ private BgpProcess createBgpProcess(RoutingInstance routingInstance) {
       IpBgpGroup ig = e.getValue();
       BgpNeighbor neighbor = new BgpNeighbor(ip, _c);
       neighbor.setVrf(vrfName);
+      String authKeyChainName = ig.getAuthKeyChainName();
+      if (ig.getAuthKeyChainName() != null) {
+        if (!_c.getAuthenticationKeyChains().containsKey(authKeyChainName)) {
+          _w.redFlag(
+              "Undefined authentication-key-chain "
+                  + authKeyChainName
+                  + "for neighbor "
+                  + ig.getRemoteAddress());
+          authKeyChainName = null;
+        } else if (ig.getAuthKey() != null) {
+          _w.redFlag(
+              "Both authentication-key and authentication-key-chain specified for neighbor "
+                  + ig.getRemoteAddress());
+        }
+      }
+      if (ig.getAuthAlgorithm() != null && ig.getAuthKey() != null) {
```

Comment:
Remove this. This will always get flagged because of inheritance of the default authentication algorithm in the master group.